src/H5Dchunk.c

```diff
@@ -2466,7 +2466,13 @@ H5D__chunk_cacheable(const H5D_io_info_t *io_info, H5D_dset_io_info_t *dset_info
     FUNC_ENTER_PACKAGE
 
     /* Sanity check */
+#ifdef H5_HAVE_PARALLEL
     HDassert(io_info);
+#else
+        /* clang-format off */
```

Comment:
wtf? Must be due to the FUNC_ENTER macro.

Reply:
This is less ugly, and I have zero interest in adding to the already-over-complicated ATTR_UNUSED scheme